src/agent/src/device.rs

```diff
@@ -393,9 +392,12 @@ fn add_device(
         return Err(anyhow!("invalid container path for device {:?}", device));
     }
 
-    match DEVICEHANDLERLIST.get(device.field_type.as_str()) {
-        None => Err(anyhow!("Unknown device type {}", device.field_type)),
-        Some(dev_handler) => dev_handler(device, spec, sandbox, devidx),
+    match device.field_type.as_str() {
+        DRIVERBLKTYPE => return virtio_blk_device_handler(device, spec, sandbox, devidx).await,
+        DRIVERMMIOBLKTYPE => return virtiommio_blk_device_handler(device, spec, sandbox, devidx),
+        DRIVERNVDIMMTYPE => return virtio_nvdimm_device_handler(device, spec, sandbox, devidx),
```

Comment:
This is @Tim-Zhang part, but I'll add my two cents :) 
- Because those two functions are not async (hence there is no `.await` tail).
Essentially we don't have to make every function async (unless we need to do async calls inside and make tokio to take care of thread management / state machine). In our case there is plain rust code with no async calls inside, so this can be regular funcs.
- I guess because we mix async and non async funcs, that might be hard to do.

Reply:
Maybe I should make these functions async for consistency reason, let me think about it.

Reply:
I have changed these functions all to async function.